### DISCUSSION TITLE
how to apply phase field to different subdomain?

### DISCUSSION THREAD

**Discussion by RWTHLHK** [idaholab/moose#29042] [Q&A Modules: Phase field]
It is very time consuming to apply phase field on an entire 3D mesh, so I only want to apply phase field to a small subdomain(block 1). But I always get material property not defined error. I know this is because I am not defining material property on different blocks in a correct way but I don't know how to fix this. Can some expert please have a look at my  inp file?
new block 1:
![image](https://github.com/user-attachments/assets/c6650301-f9b9-4d4e-b130-c34da91077af)
kernels:
![image](https://github.com/user-attachments/assets/e8ec9162-2d9e-4738-98b5-bff7a7616bb7)
materials:
![image](https://github.com/user-attachments/assets/51ed7ffb-0934-45bc-ac22-efefd8c3c768)
The error I get:
![image](https://github.com/user-attachments/assets/425fc8ac-54e4-4661-a35a-d3f242ff7bd2)

**Comment by GiudGiud** [comment]
Hello

Use `Debug/show_material_props=true` to get the names of the material properties to match up between the two equations. What is happening is that half your material properties are prefaced with `uncracked` which is set in `cracked_stress` while the rest is not prefaced, and the equations are not set consistently with one or the other. Did you block restrict the SolidMechanics physics?
